camayoc/tests/qcs/api/v1/scanjobs/test_run_scanjobs.py

```diff
@@ -303,3 +314,92 @@ def test_scan_task_results(scan_info):
         num_failed = task['systems_failed']
         num_scanned = task['systems_scanned']
         assert num_scanned == sys_count - num_failed
+
+
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_disabled_optional_products_facts(scan_info):
+    """Test scan jobs from scans with disabled optional products.
+
+    :id: 6f91ea5c-32b9-11e8-b467-0ed5f89f718b
+    :description: Test that scan jobs of scans with disabled products
+        are not gathering the raw_facts defined in the roles of the
+        disabled products.
+    :steps:
+        1) Iterate over the scans and gather the disabled_optional_products
+        2) For any scan that was defined with disabled_optional_products:
+            a) Create a facts_to_ignore list composed of raw_facts for each
+                disabled role
+            b) Iterate through the inspection results fact dictionaries
+            c) Assert that no facts that should be ignored are in the
+                dictionaries
+    :expectedresults: No facts are collected for disabled products
+    """
+    errors_found = []
+    scan = get_scan_result(scan_info['name'])
+    facts_to_ignore = []
+    disabled_optional_products = scan_info.get('disabled_optional_products')
+    if disabled_optional_products:
+        # Build the list of facts that should not be in inspection results
+        for product in disabled_optional_products.keys():
+            if product == 'jboss_eap':
+                facts_to_ignore += QCS_EAP_RAW_FACTS
+            elif product == 'jboss_fuse':
+                facts_to_ignore += QCS_FUSE_RAW_FACTS
+            elif product == 'jboss_brms':
+                facts_to_ignore += QCS_BRMS_RAW_FACTS
+        # grab the inspection results of the scan
+        inspection_results = \
+            scan.get('inspection_results')
+        for system in inspection_results:
+            fact_dicts = system.get('facts')
+            # grab the facts for each system
+            for dictionary in fact_dicts:
+                for fact in facts_to_ignore:
+                    if fact in dictionary.values():
+                        errors_found.append(
+                            'Found fact {fact_name} that should have been '
+                            'DISABLED on scan named {scan_name}'.format(
+                                fact_name=fact,
+                                scan_name=scan.get('name')))
+    assert len(errors_found) == 0, '\n================\n'.join(errors_found)
+
+
+@pytest.mark.parametrize(
```

Comment:
@kdelee I parametrized both of the functions I added, and created an errors_found list for each.